### PR TITLE
feat(service): Add Plivo service

### PR DESCRIPTION
**Pull request by prashanthpai**
Added support for Plivo as a backend service with unit tests and documentation.

### CODE REVIEW

**Comment by nikoksr** [comment]
@prashanthpai appreciate your contribution. I will look over it asap. I saw that you have mocked the service, which I find very interesting. We are having a discussion about how we can best test all services reasonably. If you are interested in sharing your experience with us, please tell us here #1.

File: service/plivo/plivo.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	}, nil
+}
+
+// AddReceivers adds the given destination phone numbers to the notifier.
+func (s *Service) AddReceivers(phoneNumbers ...string) {
+	s.destinations = append(s.destinations, phoneNumbers...)
+}
+
+// Send sends a SMS via Plivo to all previously added receivers.
+func (s *Service) Send(subject string, message string) error {
```
Comment (nikoksr) [inline service/plivo/plivo.go:78]:
```suggestion
func (s *Service) Send(subject, message string) error {
```

File: service/plivo/plivo.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+type Service struct {
+	client       plivoMsgClient
+	mopts        MessageOptions
+	destinations []string
+}
+
+// New creates a new instance of plivo service.
+func New(cOpts *ClientOptions, mOpts *MessageOptions) (*Service, error) {
+	if cOpts == nil {
+		return nil, fmt.Errorf("ClientOptions cannot be nil")
```
Comment (nikoksr) [inline service/plivo/plivo.go:44]:
```suggestion
		return nil, fmt.Errorf("client-options cannot be nil")
```

File: service/plivo/plivo.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	}
+
+	if mOpts == nil {
+		return nil, fmt.Errorf("MessageOptions cannot be nil")
```
Comment (nikoksr) [inline service/plivo/plivo.go:48]:
```suggestion
		return nil, fmt.Errorf("message-options cannot be nil")
```

**Review by nikoksr — CHANGES_REQUESTED** [review]
Just some small changes recommended by linter and we're ready to merge. Appreciate your contribution.

File: service/plivo/plivo.go (hunk 1 above)
Comment (ppai-plivo) [reply service/plivo/plivo.go]: done

File: service/plivo/plivo.go (hunk 2 above)
Comment (ppai-plivo) [reply service/plivo/plivo.go]: done

File: service/plivo/plivo.go (hunk 3 above)
Comment (ppai-plivo) [reply service/plivo/plivo.go]: done

**Review by nikoksr — APPROVED** [review]

**Pull request merged**